File: example/test1.cpp
#include <iostream>
#include <cstring>

bool isPalindrome(const char *text)
{
    int length = strlen(text);
    for (int i = 0; i < length / 2; i += 1)
    {
        if (text[i] != text[length - i - 1])
        {
            return false;
        }
    }
    return true;
}

int main()
{
    char text[100];
    printf("Enter a string: ");
    scanf("%s", text);

    if (isPalindrome(text))
    {
        printf("True");
    }
    else
    {
        printf("False");
    }

    return 0;
}

File: example/test1.py
def isPalindrome(text):
	length = len(text)
	i = 0
	while i<length/2:
		if text[i]!=text[length-i-1]:
			return False
		i+=1
	return True

def main():
	print("Enter a string: ", end='')
	text = input()
	if isPalindrome(text):
		print("True", end='')
	else:
		print("False", end='')
	return 0

if __name__ == '__main__':
	main()